### DISCUSSION TITLE
Does Vitest set a particular environment variable when a snapshot update is triggered?

### DISCUSSION THREAD

**Discussion by ffMathy** [vitest-dev/vitest#6846] [Q&A]
# Context
In our code we currently do a lot of end-to-end tests. This is because much of our logic is towards a third party system, and the main source of errors is typically in integrating with this third party system. It has all kinds of weird constraints that you won't discover if you mock it out.

But as we're expanding our test suite, it is becoming slower and slower (obviously). We want to try to find a way to strike a good balance between still having the effect of end-to-end, but faster.

As part of that, what we're thinking of doing is snapshot testing. We already do this, but we also want to have the concept of "input snapshots".

So let's say I create a test which creates a user in the third party system via an API call. We want to make it so that the first time it does this, it makes a real call, and saves the result in a snapshot file.

Then the next time, it is loaded from the snapshot file.

If someone then runs "Update snapshots" in Vitest, we want it to update the results again, as if it was the first time that this ran.

# Ask
Does Vitest provide any mechanism for detecting if "update snapshot" has been run? Does a particular environment variable get set?

**Comment by hi-ogawa** [comment]
There was a similar request on discord recently https://discord.com/channels/917386801235247114/1300883544314351677 and this is one way to do that currently using `provide/inject` https://stackblitz.com/edit/vitest-dev-vitest-get8mf?file=test%2Fglobal-setup.ts

**Reply by ffMathy** [reply]
Thank you so much! 

Is there a way to also retrieve the list of test files that are being run during global setup, or the list of test files that were run during global teardown?

This would need to work both if I just ran Vitest for all tests too (no filter).

That way I'd be able to detect if all test files were run, and also clean up my snapshots that weren't being used.

**Reply by hi-ogawa** [reply]
> Is there a way to also retrieve the list of test files that are being run during global setup, or the list of test files that were run during global teardown?

I don't think we have that in https://vitest.dev/config/#globalsetup, so it would be a feature request. 

Other than global setup, you can access `onPathsCollected` from custom reporter https://vitest.dev/guide/reporters.html#custom-reporters and you might be able to do something similar:

```ts
export default defineConfig({
  test: {
    reporters: [
      {
        onInit(ctx) {
          ctx.provide;
          ctx.config.snapshotOptions
        },
        onPathsCollected(files) {
        }
      }
    ]
  },
});
```
